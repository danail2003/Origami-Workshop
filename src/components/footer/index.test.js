import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './index';
import UserContext from '../../context';
import { Route, BrowserRouter } from 'react-router-dom';

describe('Footer', () => {
    it('should render authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: true,
                            id: '456'
                        }
                    }}>
                        <Footer />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    it('should render non-authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: false
                        }
                    }}>
                        <Footer />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });
});