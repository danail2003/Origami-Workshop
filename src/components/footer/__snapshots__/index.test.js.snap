// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render authenticated routes 1`] = `
<footer
  className="footer"
>
  <ul>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/"
        onClick={[Function]}
      >
        Publications
      </a>
    </li>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/share"
        onClick={[Function]}
      >
        Share your thoughts
      </a>
    </li>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/profile/456"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
    <img
      alt="origami"
      className="logo"
      src="blue-origami-bird-flipped.png"
    />
  </ul>
  <p
    className="paragraph"
  >
    Software University © 2019
  </p>
</footer>
`;

exports[`Footer should render non-authenticated routes 1`] = `
<footer
  className="footer"
>
  <ul>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/"
        onClick={[Function]}
      >
        Publications
      </a>
    </li>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li
      className="footer-list-item"
    >
      <a
        className="footer-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <img
      alt="origami"
      className="logo"
      src="blue-origami-bird-flipped.png"
    />
  </ul>
  <p
    className="paragraph"
  >
    Software University © 2019
  </p>
</footer>
`;
